reactive/kubernetes_master.py

```diff
@@ -1410,12 +1417,47 @@ def push_service_data():
         kube_api.configure(kubernetes_master.STANDARD_API_PORT)
 
 
-@when("certificates.available", "kube-api-endpoint.available", "cni.available")
+@when("leadership.is_leader")
+@when("endpoint.lb-provider.available")
+@when_not("kubernetes-master.sent-lb-request")
+def request_load_balancer():
+    """Request a LB from the related provider."""
+    lb_provider = endpoint_from_name("lb-provider")
+    req = lb_provider.get_request("api-server")
+    req.protocol = req.protocols.tcp
+    port = kubernetes_master.STANDARD_API_PORT
+    req.port_mapping = {port: port}
+    req.public = True
+    if not req.health_checks:
+        req.add_health_check(
+            protocol=req.protocols.http,
+            port=8080,
+            path="/livez",
+        )
+    lb_provider.send_request(req)
+    set_flag("kubernetes-master.sent-lb-request")
```

Comment:
Looks like this flag is never cleared. What if the user removes the kubernetes-master:lb-provider relaton and re-adds it? We would need to send a new request, but we never will, right?

Reply:
That was a case of me double-fixing getting duplicate health checks.  The flag technically isn't even needed, so I'll just drop it.